polkadot-protocol/protocol-components/transactions/transactions-formats.md

```diff
@@ -0,0 +1,104 @@
+
+Extrinsics normally contain a signature, some data to describe whether the extrinsic has passed some validity checks and a reference to the pallet and call that it is intended for. This format provides a way for applications to ensure that the requirements for an extrinsic are met and correctly constructed.
+
+- Unchecked - signed transactions requiring some validation check before being accepted in the transaction pool. Any unchecked extrinsic contains the signature for the sent data and some extra data
+- Checked - inherent extrinsic, which, by definition, don't require signature verification. Instead, they carry information on where the extrinsic comes from and some extra data
+- Opaque - used for cases when an extrinsic hasn't yet been committed to a format but can still be decoded
+
+Extra data can be any additional information helpful to attach to a transaction or inherent. For example, the nonce of the transaction, the tip for the block author, or how long the extrinsic is valid for. This information is provided by a specialized extensions that help determine the validity and ordering of extrinsics before they get included in a block.
+
+A signed transaction might be constructed like so:
+
+```rust
+--8<-- "code/polkadot-protocol/transactions/transactions-formats/signed-tx-example.rs"
+```
+
+## How Transactions are Constructed
+
+Polkadot SDK defines its transaction formats generically to allow developers to implement custom ways to define valid transactions. In a runtime built with FRAME, however (assuming transaction version 4), a transaction must be constructed by submitting the following encoded data:
+
+`<signing account ID> + <signature> + <additional data>`
+
+When submitting a signed transaction, the signature is constructed by signing:
+
+- The actual call is composed of the following:
+    - The index of the pallet
+    - The index of the function call in the pallet
+    - The parameters required by the function call are being targeted
+
+- Some extra information, verified by the signed extensions of the transaction:
+    - What's the era for this transaction, i.e., how long should this call last in the transaction pool before it gets discarded?
+    - The nonce, i.e., how many prior transactions have occurred from this account? This helps protect against replay attacks or accidental double-submissions
+    - The tip amount paid to the block producer to help incentivize it to include this transaction in the block
+
+Then, some additional data that's not part of what gets signed is required, which includes:
+
+- The spec version and the transaction version ensure that the transaction is submitted to a compatible runtime
```

Comment:
We could link the necessary docs for this eventually, most likely runtime upgrades which might detail what exactly these things are

Reply:
Will add a TODO:update-path so when that docs is ready, we can link it up